data_analysis/near-deduplication/minhash_deduplication_alt.py

```diff
@@ -5,697 +5,331 @@
 from __future__ import annotations
 
 import gc
-import glob
 import hashlib
-import json
 import logging
-import multiprocessing
+import multiprocessing as mp
 import os
 import random
 import re
+import struct
 import time
 import warnings
+from collections import defaultdict
+from itertools import tee
 from pathlib import Path
-from typing import Any, Dict, Iterable, List, Set
-
-warnings.filterwarnings("ignore", category=FutureWarning)
-multiprocessing.set_start_method("fork", force=True)
-
-import datasets
-import dill as pickle
-import networkit as nk
-import numpy as np
-import typer
-from datasets import Dataset, concatenate_datasets, load_dataset, load_from_disk
-from datasketch import LeanMinHash, MinHash, MinHashLSH
-from rich.console import Console
-from rich.logging import RichHandler
-from tqdm import tqdm
-
-random.seed(42)
-MINHASH_SEED = 42
+from typing import Any
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Tuple
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=FutureWarning)
+    import datasets
```

Comment:
are those imports intentionally indented? If so maybe add comment explaining this?

Reply:
There is a `with` statement above to filter warnings from 3rd party imports, so yes, they have to be indented.